### PR TITLE
Makefile: Install and library targets and configurable BUILDDIR

### PR DESCRIPTION
**Pull request by zuiderkwast**
* Configurable BUILDDIR
* Add install and library targets
* Remove unnecessary variables

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,6 +20,14 @@
 
 HELPER ?=
 BINEXT ?=
+BINDIR ?= $(CURDIR)
```
Comment (bjosv) [inline Makefile]: I've seen that` BUILDDIR` has been proposed in this old exchange:
https://github.com/nodejs/http-parser/issues/418

I thought `BINDIR` was used to describe where to install binaries, but there might be different religions about this..

**Review by bjosv — APPROVED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (zuiderkwast) [reply Makefile]: I don't know. Maybe BUILDDIR is better, if you've seen it in other Makefiles.

**Review by bjosv — COMMENTED** [review]
WDYT about renaming SOLIBNAME to LIBNAME, and SOLIBNAME to LIBNAME?
i.e
```
LIBNAME = liburl_parser
SONAME ?= $(LIBNAME).$(SOEXT).$(SOMAJOR).$(SOMINOR)
SOLIBNAME= $(LIBNAME).$(SOEXT).$(SOMAJOR).$(SOMINOR).$(SOREV)
```
Then the LIBNAME can additionally be used for the static library.. or maybe we divert to much from the original then?

**Comment by zuiderkwast** [comment]
> WDYT about renaming SOLIBNAME to LIBNAME, and SOLIBNAME to LIBNAME?

Yes we can. It makes sense.

> or maybe we divert to much from the original then?

Yes, that's why i didn't do the change. But it doesn't matter much. It's more important than the C code is as close as possible to the original. I'll do the change.

**Comment by zuiderkwast** [comment]
Looks better? Do you want a variable for `$(LIBNAME).a`?

**Review by bjosv — APPROVED** [review]
I'm happy!

**Pull request merged**